src/types.typ

```diff
@@ -8,11 +8,32 @@
 #import "types/string.typ": string, ip, email
 #import "types/tuple.typ": tuple
 
+#let alignment = base-type.with(name: "alignment", types: (alignment,))
+#let angle = base-type.with(name: "angle", types: (angle,))
 #let any = base-type.with(name: "any")
 #let boolean = base-type.with(name: "bool", types: (bool,))
+#let bytes = base-type.with(name: "bytes", types: (bytes,))
 #let color = base-type.with(name: "color", types: (color,))
 #let content = base-type.with(name: "content", types: (content, str))
 #let date = base-type.with(name: "date", types: (datetime,))
+#let direction = base-type.with(name: "direction", types: (direction,))
+#let function = base-type.with(name: "function", types: (function,))
+#let fraction = base-type.with(name: "fraction", types: (fraction,))
+#let gradient = base-type.with(name: "gradient", types: (gradient,))
+#let label = base-type.with(name: "label", types: (label,))
+#let length = base-type.with(name: "length", types: (length,))
+#let location = base-type.with(name: "location", types: (location,))
+#let plugin = base-type.with(name: "plugin", types: (plugin,))
+#let ratio = base-type.with(name: "ratio", types: (ratio,))
+#let regex = base-type.with(name: "regex", types: (regex,))
+#let relative = base-type.with(
+  name: "relative",
+  types: (relative, ratio, length),
+)
```

Comment:
I'm a little concerned about this one, `length` and `ratio` will not have the same methods as `relative` for example, `z.parse` would return whatever type is passed without converting it to a relative.

This can be problematic, as shown [here](https://github.com/tingerrr/hydra/blob/d6e869fa0d78b5997ac16707e468b36ef97b537d/src/core.typ#L45-L46).

Should we be less permissive about the types or add a default post-transformation?